DataTypes/Car.cs

```diff
@@ -87,6 +87,11 @@ public bool Equals(Car other)
 
         // required for serialization to work
         public Car() { }
+
+        public override bool Equals(object obj)
+        {
+            return Equals(obj as Car);
```

Comment:
Shouldn't equal method compare all relevent parameters and the return true/false based on that?

Reply:
Car is identified by its ID, so I thought that since no two different cars can have the same ID, I should not waste time comparing other properties

Reply:
I think it makes sense to compare by ID only, atleast in this case. Seems like we won't really compare vehicles based on their paremeters just yet (or atleast won't override Equals for this cause)

Reply:
What if there's an error where two vehicles have the same ID, but different params?

Reply:
If this happens, server should crash (at least in my opinion), because it isn't working correctly

Reply:
Under no circumstances should the server crash. It should throw an exception something like DuplicateIDException and handle this behavior to avoid crashing.